src/olympia/abuse/actions.py

```diff
@@ -610,12 +624,34 @@ class ContentActionTargetAppealApprove(
 ):
     description = 'Reported content is within policy, after appeal'
 
+    @property
+    def target_versions(self):
+        target = self.target
+        if isinstance(target, Addon) and target.status == amo.STATUS_DISABLED:
+            qs = target.versions(
+                manager='unfiltered_for_relations'
+            ).disabled_that_would_be_renabled_with_addon()
```

Comment:
Is there an alternative that doesn't involve us adding another queryset method to VersionManager that is only used here, and is a duplicate of the same method on FileManager?
`File.objects.filter(version__addon=target).disabled_that_would_be_renabled_with_addon().values_list('version', flat=True)`?

Reply:
Or if we're worried about the extra overhead of loading full-fat Version instances, return `version__version` and `version_id` and send `*target_versions` as `*((Version, version_id) for _, version_id in target_versions)` instead.  (Because the underlying ActivityLog.create can accept (Class, pk) tuples rather than instances)

Reply:
I wanted to stay as consistent as possible (which meant returning `Version`s) with the rest of the code while at the same time using a query that performs well and loads as little as possible because there could be thousands of versions.

I'll play around with it and try to remove the extra method, I also don't like it

Reply:
I figured, but you don't need to have a `target_versions` method on this class - it could be called something different, or the query inlined elsewhere

Reply:
Can't inline completely, the main issue is that the query is used in 2 places:
- Here, in the `ContentActionTargetAppealApprove`
- In `reviewers/utils.py`'s `ReviewHelper.enable_addon()`

But I've pushed 2c952700a2d618c2a450a1c508576a134511d5f4 which should be good enough and removes the duplicate method